Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -955,13 +889,23 @@ public bool IsNewOrModified(QualifiedModuleName key)
             if (_moduleStates.TryGetValue(key, out var moduleState))
             {
                 // existing/modified
-                return moduleState.IsNew || key.ContentHashCode != moduleState.ModuleContentHashCode;
+                return moduleState.IsNew || GetModuleContentHash(key) != moduleState.ModuleContentHashCode;
             }
 
             // new
             return true;
         }
 
+        private int GetModuleContentHash(QualifiedModuleName module)
+        {
+            using (var codeModule = ProjectsProvider.Component(module).CodeModule)
+            {
+                return codeModule != null && codeModule.CountOfLines > 0
+                        ? codeModule.GetLines(1, codeModule.CountOfLines).GetHashCode()
```

Comment:
why `GetHashCode() `? I'd have expected `Count` or `Length` or something like that...

Reply:
I just copied what `QualifiedModuleName` was using before, for which this is the replacement.

Reply:
This totally makes semse since `string` overloads `GetHashCode` to match its `Equals` method. So, comparing the results of `GetHashCode` is a good approximation of equality. 
Only comparing the length or the count would not detect many cases of changed modules.

Reply:
Don't we have an actual `ContentHash` that would be less prone to hash collisions? Not that it's anywhere near reasonably likely, just curious.

Reply:
Just wanted to write this. The real question is why we never used `ContentHash` from `ICodeModule`, which has been implemented in a different way. Should I change it?

Reply:
If the goal is to determine whether the code was changed, yeah - that's exactly why `ContentHash` exists.